### PR TITLE
Do not automatically parse rational constants from NEG and DIVISION

### PR DESCRIPTION
**Pull request by ajreynol**
This eliminates an optimization in our parser where terms that specific rational or numeral values are automatically.

This increases our proof size by 0.1% (985k steps vs. 984k steps) and our number of trusted rewrite steps by 0.08% (135742 vs 135632).

This includes https://github.com/cvc5/cvc5/pull/10471, review that PR first.

Some benchmarks that used array constants required changing. In particular `((as const (Array Real Real)) (/ 1 4))` is no longer a valid term, since `(/ 1 4)` is the division of 1 and 4 and not a rational value.  Due to our partial, experimental support of array constants that only accept values, the *required* syntax for this is now `((as const (Array Real Real)) 1/4)`.

FYI @barrettcw 

This PR also introduces `--print-arith-lit-token` (disabled by default) which prints rational values and negative arithmetic values using the new syntax.

Based on these changes, it simplifies and improves the ALF printer.

### CODE REVIEW

File: src/parser/smt2/smt2_term_parser.cpp [hunk 1]
```diff
@@ -284,6 +284,11 @@ Term Smt2TermParser::parseTerm()
         ret = tm.mkReal(d_lex.tokenStr());
       }
       break;
+      case Token::RATIONAL_LITERAL:
+      {
+        ret = d_state.getSolver()->mkReal(d_lex.tokenStr());
```
Comment (aniemetz) [inline src/parser/smt2/smt2_term_parser.cpp:289]: Use term manager (also one of the reasons why CI fails).

File: src/parser/smt2/smt2_state.cpp [hunk 2]
```diff
@@ -1536,31 +1536,10 @@ Term Smt2State::applyParseOp(const ParseOp& p, std::vector<Term>& args)
...
-        // (- n) denotes a negative value
-        std::stringstream suminus;
-        suminus << "-" << args[0].getIntegerValue();
-        Term ret = d_tm.mkInteger(suminus.str());
-        Trace("parser") << "applyParseOp: return negative constant " << ret
-                        << std::endl;
-        return ret;
-      }
-      Term ret = d_tm.mkTerm(Kind::NEG, {args[0]});
+      Term ret = d_solver->mkTerm(Kind::NEG, {args[0]});
```
Comment (aniemetz) [inline src/parser/smt2/smt2_state.cpp:1539]: Use term manager (also one of the reasons why CI fails).

File: src/printer/smt2/smt2_printer.cpp [hunk 3]
```diff
@@ -73,14 +73,22 @@ static void toStreamRational(std::ostream& out,
...
+  bool arithTokens = options::ioutils::getPrintArithLitToken(out);
   // Print the rational, possibly as a real.
   // Notice that we print (/ (- 5) 3) instead of (- (/ 5 3)),
   // the former is compliant with real values in the smt lib standard.
   if (r.isIntegral())
   {
     if (neg)
     {
-      out << "(- " << -r;
+      if (arithTokens)
```
Comment (abdoo8080) [inline src/printer/smt2/smt2_printer.cpp:84]: Too many `arithTokens` checks. Could you reduce them to just 2 checks: one for integers and one for rationals?

**Review by abdoo8080 — CHANGES_REQUESTED** [review]

File: src/printer/smt2/smt2_printer.cpp (hunk 3 above)
Comment (ajreynol) [reply src/printer/smt2/smt2_printer.cpp]: Sure, this is a bit more verbose but I've updated now.

**Review by abdoo8080 — APPROVED** [review]

**Pull request merged**